source/tutorials/learning-journey/sharing-dependencies.md

```diff
@@ -0,0 +1,198 @@
+# Sharing dependencies between `default.nix` and `shell.nix`
+
+<!-- Include any foreward you want here -->
+
+## Overview
+
+### What will you learn?
+
+In this tutorial you'll learn how not to repeat yourself by sharing dependencies between 'default.nix`, which is responsible for building the project, and `shell.nix`, which is responsible for providing you an environment to work in.
+
+### How long will it take?
+
+This tutorial will take approximately 1 hour.
+
+### What will you need?
+
+This tutorial assumes you've seen a derivation (`mkDerivation`, `buildPythonApplication`, etc) before, and that you've seen `nix-shell` used to create shell environments.
+While this tutorial uses Python as the language for the example project, no actual Python knowledge is requried.
+
+## Setting the stage
+Suppose you have a working build for your project in a `default.nix` file so that when you type `nix-build` in your shell it builds your project.
+It includes all of the dependencies needed to build it, but nothing more.
+Now suppose you wanted to bring in some tools during development, such as a linter, a code formatter, etc.
```

Comment:
Git commit hooks is a good one to mention (e.g. https://github.com/cachix/pre-commit-hooks.nix)